File: 1470.Shuffle the Array.cc
// https://leetcode.com/problems/shuffle-the-array/submissions/

static bool x=[](){
    std::ios::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);
    return true;
}();

class Solution {
public:
    vector<int> shuffle(vector<int>& nums, int n) {
        vector<int> shuffled(nums.size());
        shuffled.clear();
        shuffled.push_back(nums[0]);

        for (unsigned i = 1, j = n; i != n && j < nums.size(); ++i, ++j)
        {
            shuffled.push_back(nums[j]);
            shuffled.push_back(nums[i]);
        }
        shuffled.push_back(nums[nums.size() - 1]);

        return move(shuffled);
    }
};

File: leetcode/1201-1500/1451-1500/1470.Shuffle the Array.rs
impl Solution {
    pub fn shuffle(nums: Vec<i32>, n: i32) -> Vec<i32> {
        let mut shuffled = Vec::with_capacity(nums.len());
        for idx in 0..n {
            shuffled.push(nums[idx as usize]);
            shuffled.push(nums[(idx + n) as usize]);
        }
        shuffled
    }
}